test/gtest/ucm/cuda_hooks.cc

```diff
@@ -25,6 +25,18 @@ static void cuda_mem_free_callback(ucm_event_type_t event_type,
     free_event.mem_type.mem_type = event->mem_type.mem_type;
 }
 
+static void check_cuda_malloc_async_capable() {
+    int supported_cuda_version;
+    CUresult cu_err;
+
+    cu_err = cuDriverGetVersion(&supported_cuda_version);
+    ASSERT_EQ(cu_err, CUDA_SUCCESS);
+
+    if (supported_cuda_version < 11020) {
+		GTEST_SKIP_("cuMemAllocAsync not available");
+    }
```

Comment:
Originally I tried to do this with `cuGetProcAddress`, but `cuGetProcAddress` only got introduced in CUDA 11.3. I guess we could skip tests for that version as well but that feels like introducing an artificial cap, so checking the latest CUDA version supported by the driver seems like a better alternative here.